probe-rs/src/bin/probe-rs/main.rs

```diff
@@ -215,18 +215,19 @@ fn prune_logs(directory: &Path) -> Result<(), anyhow::Error> {
     Ok(())
 }
 
-/// Returns the cleaned arguments for the handler of the respective end binary (cli, cargo-flash, cargo-embed, etc.).
-fn multicall_check(args: &[OsString], want: &str) -> Option<Vec<OsString>> {
+/// Returns the cleaned arguments for the handler of the respective end binary
+/// (cli, cargo-flash, cargo-embed, etc.), without the executable name.
+fn multicall_check<'list>(args: &'list [OsString], want: &str) -> Option<&'list [OsString]> {
     let argv0 = Path::new(&args[0]);
     if let Some(command) = argv0.file_stem().and_then(|f| f.to_str()) {
         if command == want {
-            return Some(args.to_vec());
+            return Some(&args[1..]);
```

Comment:
In what cases can this first branch actually match? If I see correctly, this can only happen if the user renames the `probe-rs` executable to either `cargo-ember` or `cargo-flash`, which is... probably not intended?

Reply:
This seems like a good place to add some tests, that would make it clear how this function should work.

I find it hard to see right now what the input arguments for this would look like, in combination with the preprocessing in `cargo-embed` and `cargo-flash`.

Reply:
That is fair and it also made me realize I broke clap. However, now I question the validity of some of this PR because of this.